userspace/libsinsp/filterchecks.cpp

```diff
@@ -3891,6 +3891,17 @@ uint8_t* sinsp_filter_check_event::extract(sinsp_evt *evt, OUT uint32_t* len, bo
 				ASSERT(parinfo->m_len == sizeof(uint16_t));
 				uint16_t evid = *(uint16_t *)parinfo->m_val;
 
+				// Only generic enter event has the nativeID as second param
+				if (m_inspector->is_capture() && evid == PPM_SC_UNKNOWN && etype == PPME_GENERIC_E)
+				{
+					// try to enforce a forward compatibility for syscalls added
+					// after a scap file was generated,
+					// by looking up using nativeID
```

Comment:
This only works for same-architecture scap file -> replay, of course. 
Better than nothing i guess.

Reply:
yeah I would add a comment to highlight this detail of the same architecture